### PR TITLE
Receiving chain incorrectly checks for the owner of nft when sending chain is not own chain

### LINKED ISSUE #8831: Receiving chain incorrectly checks for the owner of nft when sending chain is not own chain
### Expected behavior
Receiving chain should check for the owner of nft only when sending chain is own chain
### Actual behavior
Receiving chain [checks](https://github.com/LiskHQ/lisk-sdk/blob/feature/6917-implement-nft-module/framework/src/modules/nft/cc_commands/cc_transfer.ts#L77) for the owner of nft even when sending chain is not own chain
### Steps to reproduce
Transfer nft b/w two different chains and observe the logs
### Which version(s) does this affect? (Environment, OS, etc...)
feature/6917-implement-nft-module

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #8831 

### How was it solved?

Update verify logic of CrossChainTransferCommand in nft module

### How was it tested?

- Added a new test case
- Updated unit test
- Transfer nft b/w two different chains and observe the logs

### CODE REVIEW

**Comment by shuse2** [comment]
@Incede I think it would be better if we add test for this case

**Comment by Incede** [comment]
> @Incede I think it would be better if we add test for this case

Added a new test case.

Additionaly, while adding the new test case I figured the check if nft exists is repeated 2 times, immediately one after another with different error messages so I think it makes sense to remove one of them. Please lmk if you have a different opinion.

File: framework/src/modules/nft/cc_commands/cc_transfer.ts [hunk 1]
```diff
@@ -70,13 +70,12 @@ export class CrossChainTransferCommand extends BaseCCCommand {
...
 		const nftExists = await nftStore.has(getMethodContext(), nftID);
-		if (nftChainID.equals(ownChainID) && !nftExists) {
-			throw new Error('Non-existent entry in the NFT substore');
-		}
 
-		const owner = await this._method.getNFTOwner(getMethodContext(), nftID);
-		if (nftChainID.equals(ownChainID) && !owner.equals(sendingChainID)) {
-			throw new Error('NFT has not been properly escrowed');
+		if (nftChainID.equals(ownChainID)) {
+			const owner = await this._method.getNFTOwner(getMethodContext(), nftID);
```
Comment (has5aan) [inline framework/src/modules/nft/cc_commands/cc_transfer.ts:75]: Should an update in LIP be requested as well? It suggests `raise Exception("Non-existent entry in the NFT substore")` which is not consistent with other places in LIP that suggests to throw `NFT substore entry does not exist` for the scenario.

**Review by has5aan — APPROVED** [review]

File: framework/src/modules/nft/cc_commands/cc_transfer.ts (hunk 1 above)
Comment (Incede) [reply framework/src/modules/nft/cc_commands/cc_transfer.ts:75]: probably yes

**Review by shuse2 — APPROVED** [review]

**Pull request merged**